src/processor/operator/persistent/reader/csv/base_csv_reader.cpp

```diff
@@ -369,6 +370,10 @@ uint64_t BaseCSVReader::parseCSV(Driver& driver) {
         position++;
         do {
             for (; position < bufferSize; position++) {
+                if (typeid(driver) == typeid(SniffCSVDialectDriver)) {
```

Comment:
Instead of using typeid to check the type of each driver, i would suggest adding a new field `DriverType` in the driver base class.